File: polyglot/seis_native/csharp/SeisNativeKernel.cs
using System.Collections.Generic;
using System.Linq;

namespace Seis.NativeKernel;

public record NativeRoadmapItem(string Lane, int Score);

public static class SeisNativeKernel
{
    public const string Version = "2026.07.1";

    public static IReadOnlyList<NativeRoadmapItem> Roadmap { get; } = new List<NativeRoadmapItem>
    {
        new("Apple First", 100),
        new("Data AI", 88),
        new("Systems", 84),
        new("Android", 76),
        new("Windows", 72),
        new("Infrastructure", 70)
    };

    public static IReadOnlyList<NativeRoadmapItem> OrderedRoadmap() =>
        Roadmap.OrderByDescending(item => item.Score).ToList();

    public static NativeRoadmapItem TopLane() => OrderedRoadmap().First();
}

File: polyglot/seis_native/elixir/seis_native_kernel.ex
defmodule SeisNativeKernel do
  @lanes [
    %{lane: "Apple First", score: 100},
    %{lane: "Data AI", score: 88},
    %{lane: "Systems", score: 84},
    %{lane: "Android", score: 76},
    %{lane: "Windows", score: 72},
    %{lane: "Infrastructure", score: 70}
  ]

  def lanes, do: @lanes

  def top_lane, do: hd(@lanes)
end
